tools/validate_spec.py

```diff
@@ -101,6 +110,32 @@ def cli(argv: List[str]) -> Union[int, str]:
     except RecordFluxError as e:
         return f"invalid identifier: {e}"
 
+    checksum_functions = None
+    if args.checksum_functions is not None:
+        try:
+            checksum_module = importlib.import_module(str(args.checksum_functions))
+        except ImportError as e:
+            return (
+                f"The provided module {args.checksum_functions} cannot be "
+                f"imported. Make sure the module name is provided as "
+                f"package.module and not as a file system path. {e}"
+            )
+
+        try:
+            all_checksum_functions = checksum_module.checksum_functions  # type: ignore
+        except AttributeError:
+            return (
+                f"The checksum module at {args.checksum_functions} "
+                f'does not contain a dict with the name "checksum_function". '
+                "Provide a dict that maps the message identifier to a dict "
+                "that maps the name of each checksum field to a callable "
+                "checksum function."
+            )
+        try:
+            checksum_functions = all_checksum_functions[args.message_identifier]
+        except KeyError:
+            return f"The checksum_function dict does not contain a key for " f"{identifier}"
```

Comment:
It is not checked that `checksum_functions` is a `dict` and that the values of the dict are functions.

Reply:
Added in 66a82d8c (the commit contains two duplicate tests. the duplicates were removed in 587ec6c2)